content/sensu-go/6.8/operations/deploy-sensu/secure-postgres.md

```diff
@@ -0,0 +1,376 @@
+**NOTE**: The Sensu backend uses the libpq library to make connections to PostgreSQL.
+This library [supports a number of environment variables](https://www.postgresql.org/docs/current/libpq-envars.html) that can be injected into the PostgreSQL data source name (DSN) and are loaded at runtime using the system's environment variable file.
+{{% /notice %}}
+
+Working from your Sensu backend, follow these steps to configure Sensu to use certificate authentication with PostgreSQL:
+
+1. Define the environment variables that tell the Sensu backend to use a certificate to authenticate to PostgreSQL:
+
+   {{< language-toggle >}}
+
+{{< code "RHEL/CentOS" >}}
+echo 'PGUSER=sensu
+PGSSLMODE="verify-full"
+PGSSLCERT="/etc/sensu/tls/sensu.pem"
+PGSSLKEY="/etc/sensu/tls/sensu-key.pem"
+PGSSLROOTCERT="/etc/sensu/tls/ca.pem"' | sudo tee /etc/sysconfig/sensu-backend
+{{< /code >}}
+
+{{< code "Ubuntu/Debian" >}}
+echo 'PGUSER=sensu
+PGSSLMODE="verify-full"
+PGSSLCERT="/etc/sensu/tls/sensu.pem"
+PGSSLKEY="/etc/sensu/tls/sensu-key.pem"
+PGSSLROOTCERT="/etc/sensu/tls/ca.pem"' | sudo tee /etc/default/sensu-backend
+{{< /code >}}
+
+{{< /language-toggle >}}
+
+2. Adjust the Sensu datastore connection with sensuctl:
+
+   {{< code shell >}}
+
+echo 'type: PostgresConfig
+api_version: store/v1
+metadata:
+  name: sensu_postgres
+spec:
+  dsn: "postgresql://postgres.example.com:5432/sensu_events"
+  pool_size: 20
+  strict: false' | sudo tee postgresconfig.yml
+
+sensuctl create -f postgresconfig.yml
+
+{{< /code >}}
+
+   {{% notice note %}}
+**NOTE**: Setting `strict: false` in the configuration helps ensure that the Sensu backend will remain active and able to process events even in case of a configuration mistake.
+{{% /notice %}}
+
+3. Confirm that the connection to your PostgreSQL instance is healthy:
+
+   {{< code shell >}}
+curl http://localhost:8080/health
+{{< /code >}}
+
+   The response should be similar to this example, with `true` values for both `Active` and `Healthy`:
+
+   {{< code text >}}
+{
+  "Alarms": null,
+  "ClusterHealth": [
+    {
+      "MemberID": 13217573501179607000,
+      "MemberIDHex": "b76e4111d26d35e2",
+      "Name": "sensu.example.com",
+      "Err": "",
+      "Healthy": true
+    }
+  ],
+  "Header": {
+    "cluster_id": 11959078708079102000,
+    "member_id": 6370351775894128000,
+    "raft_term": 4242
+  },
+  "PostgresHealth": [
+    {
+      "Name": "sensu_postgres",
+      "Active": true,
+      "Healthy": true
+    }
+  ]
+}
+{{< /code >}}
+
+Now that you've confirmed that the Sensu backend can connect to your PostgreSQL instance, you can configure PostgreSQL to use TLS.
+
+### Configure PostgreSQL to use TLS
+
+To configure your PostgreSQL instance to use TLS:
+
+1. Copy your PostgreSQL certificate files from your Sensu backend.
+From the `/etc/sensu/tls` directory, run:
+
+   {{< code shell >}}
+scp postgres.example.com* postgres.example.com:/home/user
+scp ca.pem postgres.example.com:/home/user
+{{< /code >}}
```

Comment:
I get the following responses to these commands:

```
[root@sensu-centos tls]# scp postgres.example.com* postgres.example.com:/home/user
ssh: Could not resolve hostname postgres.example.com: Name or service not known
lost connection
[root@sensu-centos tls]# scp ca.pem postgres.example.com:/home/user
ssh: Could not resolve hostname postgres.example.com: Name or service not known
lost connection
```

Reply:
@hillaryfraley that's sort of expected. You'll have to edit that command for your postgres instance.

Reply:
oh I see, the home/user part... doh

Reply:
No worries!